engine/class_modules/warlock/sc_warlock.cpp

```diff
@@ -55,6 +55,73 @@ struct drain_life_t : public warlock_spell_t
   }
 };
 
+struct decimating_bolt_dmg_t : public warlock_spell_t
+{
+  decimating_bolt_dmg_t( warlock_t* p ) : warlock_spell_t( "decimating_bolt_tick_t", p, p->find_spell( 327059 ) )
+  {
+    background = true;
+    may_miss   = false;
+    dual       = true;
+  }
+
+  double composite_target_multiplier( player_t* target ) const override
+  {
+    double m = warlock_spell_t::composite_target_multiplier( target );
+
+    m *= 2.0 - target->health_percentage() * 0.01;
+
+    return m;
+  };
+
+  void impact( action_state_t* s ) override
+  {
+    warlock_spell_t::impact( s );
+  };
+
+
+};
+
+struct decimating_bolt_t : public warlock_spell_t
+{
+  action_t* decimating_bolt_dmg;
+
+  decimating_bolt_t( warlock_t* p, util::string_view options_str ) : 
+    warlock_spell_t( "decimating_bolt", p, p->covenant.decimating_bolt ),
+    decimating_bolt_dmg( new decimating_bolt_dmg_t( p ) )
+
+  {
+    parse_options( options_str );
+    // can_havoc = true; NYI
+    travel_speed = p->find_spell( 327072 )->missile_speed();
+
+    add_child( decimating_bolt_dmg );
+  }
+
+  void impact( action_state_t* s ) override
+  {
+    if ( p()->talents.fire_and_brimstone->ok() )
```

Comment:
Instead of having 2 lines triggering the buff I would only calculate the parameter you pass to the trigger function depending on whether you have FaB.
That way it is easier to understand that the buff always gets triggered, just with different values.

Reply:
I do agree it's a bit verbose, but creating a trigger function doesn't seem necessary for this one off situation. I believe a middle ground would be to create shared variable that changes based off an if condition. This would shorten it down and still maintain the readability.

Reply:
> Instead of having 2 lines triggering the buff I would only calculate the parameter you pass to the trigger function depending on whether you have FaB.
> That way it is easier to understand that the buff always gets triggered, just with different values.

I do agree that it's a bit verbose, but creating a trigger function doesn't seem necessary for this one off situation. I believe a middle ground would be to create shared variable that changes based off an if condition. This would shorten it down and still maintain the readability.

Reply:
Yeah that's what I meant, I just wrote it a bit confusingly.